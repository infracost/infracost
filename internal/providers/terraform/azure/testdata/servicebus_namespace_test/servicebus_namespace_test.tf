provider "azurerm" {
  skip_provider_registration = true
  features {}
}

resource "azurerm_resource_group" "example" {
  name     = "example-resources"
  location = "West Europe"
}

locals {
  options = [
    {
      sku      = ["Basic", "Standard"]
      capacity = [0]
    },
    {
      sku      = ["Premium"]
      capacity = [1, 2, 4, 8, 16]
    },
  ]

  permutations = distinct(flatten([
    for option in local.options : [
      for sku in option.sku : [
        for capacity in option.capacity : {
          sku      = sku
          capacity = capacity
        }
      ]
    ]
  ]))
}

resource "azurerm_servicebus_namespace" "servicebus_namespace" {
  for_each = { for entry in local.permutations : "${entry.sku}.${entry.capacity}" => entry }

  name                = "servicebus-namespace"
  resource_group_name = azurerm_resource_group.example.name
  location            = azurerm_resource_group.example.location

  sku      = each.value.sku
  capacity = each.value.capacity
}

resource "azurerm_servicebus_namespace" "servicebus_namespace_with_usage" {
  for_each = { for entry in local.permutations : "${entry.sku}.${entry.capacity}" => entry }

  name                = "servicebus-namespace-with-usage"
  resource_group_name = azurerm_resource_group.example.name
  location            = azurerm_resource_group.example.location

  sku      = each.value.sku
  capacity = each.value.capacity
}
