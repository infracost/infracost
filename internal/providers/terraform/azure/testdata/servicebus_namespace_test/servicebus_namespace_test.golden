
 Name                                                                              Monthly Qty  Unit                    Monthly Cost 
                                                                                                                                     
 azurerm_servicebus_namespace.servicebus_namespace["Basic.0"]                                                                        
 └─ Messaging operations                                                     Monthly cost depends on usage: $0.05 per 1M operations  
                                                                                                                                     
 azurerm_servicebus_namespace.servicebus_namespace["Premium.1"]                                                                      
 └─ Messaging units                                                                          1  units                        $677.08 
                                                                                                                                     
 azurerm_servicebus_namespace.servicebus_namespace["Premium.16"]                                                                     
 └─ Messaging units                                                                         16  units                     $10,833.20 
                                                                                                                                     
 azurerm_servicebus_namespace.servicebus_namespace["Premium.2"]                                                                      
 └─ Messaging units                                                                          2  units                      $1,354.15 
                                                                                                                                     
 azurerm_servicebus_namespace.servicebus_namespace["Premium.4"]                                                                      
 └─ Messaging units                                                                          4  units                      $2,708.30 
                                                                                                                                     
 azurerm_servicebus_namespace.servicebus_namespace["Premium.8"]                                                                      
 └─ Messaging units                                                                          8  units                      $5,416.60 
                                                                                                                                     
 azurerm_servicebus_namespace.servicebus_namespace["Standard.0"]                                                                     
 ├─ Base charge                                                                            730  hours                          $9.81 
 ├─ Messaging operations (13M-100M)                                          Monthly cost depends on usage: $0.80 per 1M operations  
 └─ Brokered connections (1K-100K)                                           Monthly cost depends on usage: $0.03 per connections    
                                                                                                                                     
 azurerm_servicebus_namespace.servicebus_namespace_with_usage["Basic.0"]                                                             
 └─ Messaging operations                                                                 1,000  1M operations                 $50.00 
                                                                                                                                     
 azurerm_servicebus_namespace.servicebus_namespace_with_usage["Premium.1"]                                                           
 └─ Messaging units                                                                          1  units                        $677.08 
                                                                                                                                     
 azurerm_servicebus_namespace.servicebus_namespace_with_usage["Premium.16"]                                                          
 └─ Messaging units                                                                         16  units                     $10,833.20 
                                                                                                                                     
 azurerm_servicebus_namespace.servicebus_namespace_with_usage["Premium.2"]                                                           
 └─ Messaging units                                                                          2  units                      $1,354.15 
                                                                                                                                     
 azurerm_servicebus_namespace.servicebus_namespace_with_usage["Premium.4"]                                                           
 └─ Messaging units                                                                          4  units                      $2,708.30 
                                                                                                                                     
 azurerm_servicebus_namespace.servicebus_namespace_with_usage["Premium.8"]                                                           
 └─ Messaging units                                                                          8  units                      $5,416.60 
                                                                                                                                     
 azurerm_servicebus_namespace.servicebus_namespace_with_usage["Standard.0"]                                                          
 ├─ Base charge                                                                            730  hours                          $9.81 
 ├─ Messaging operations (13M-100M)                                                         87  1M operations                 $69.60 
 ├─ Messaging operations (100M-2,500M)                                                   2,400  1M operations              $1,200.00 
 ├─ Messaging operations (over 2,500M)                                                     500  1M operations                $100.00 
 ├─ Brokered connections (1K-100K)                                                      99,000  connections                $2,970.00 
 ├─ Brokered connections (100K-500K)                                                   400,000  connections                $6,000.00 
 └─ Brokered connections (over 500K)                                                   500,000  connections               $12,500.00 
                                                                                                                                     
 OVERALL TOTAL                                                                                                            $64,887.87 
──────────────────────────────────
15 cloud resources were detected:
∙ 14 were estimated
∙ 1 was free:
  ∙ 1 x azurerm_resource_group

┏━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━┳━━━━━━━━━━━━━━┓
┃ Project                                            ┃ Monthly cost ┃
┣━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━╋━━━━━━━━━━━━━━┫
┃ TestServiceBusNamespace                            ┃ $64,888      ┃
┗━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━┻━━━━━━━━━━━━━━┛